### PR TITLE
Introducing the Ion Carbine!

### PR DESCRIPTION
**Pull request by GunHog**
- Adds the Ion Carbine, a compact version of the Ion Rifle that users
  can carry on their belt or in a regular backpack.
- Adds the Ion Carbine to R&D, replacing the Ion Rifle research option.
- Ion Rifle now starts with a pin installed as it can no longer be made
  in Science.
- The Ion Rifle and Ion Carbine can now accept Seclites as an
  attachment.

<b>Left: Ion Carbine, without Seclite, Seclite off, Seclite On. Right: Ion Rifles.</b>
screenshot 2015-02-10 11 57 47

<b>Belt sprite, for when attached to belt!</b>
screenshot 2015-02-10 16 45 59

<b>Comparison between rifle and carbine inhands</b>
screenshot 2015-02-12 09 14 58

### CODE REVIEW

**Comment by paprka** [comment]
God damnit dude are you blind

**Comment by GunHog** [comment]
Maybe I am. I will revisit this once your stuff is merged.

**Comment by GunHog** [comment]
On second thought, I will leave this one open so people can look over it. Do not merge until #7651 is merge and this one is resolved for its conflicts, or if that one is closed.

**Comment by Lumipharon** [comment]
Please no, emp rifles are super strong already. If you want to be able to store one, you already have the option of getting a BoH anyway.

**Comment by paprka** [comment]
Those don't fit ion rifles i'm pretty sure

**Comment by TZK13** [comment]
They do fit, as far as I know there's not a single thing that you can hold in your hands that can't fit inside a BoH.

Anyways I support this, there's no reason why ion weapons should be as limited as they are.

**Comment by Dorsisdwarf** [comment]
As a silicon player I agree that science needs some late-game wreck-their-shit tools.

Even if X-Rays kinda fulfill that role for rogue AIs anyway.

In either case, I like this.

**Comment by optimumtact** [comment]
>EMP rifles are super strong

Wat

**Comment by Lumipharon** [comment]
>shoot a sec officer once
All ranged weapons are now dead.
Headset is fried so no calling for help. 
Flash is dead, good luck dealing with that emagged borg.
It's AOE, so not only can you take out multiple people with one shot, you can do so around corners etc.
You can also use it to shock/open doors/take down cameras, trigger firelocks, and of course wreck borgs and mechs.

It's not OP right now BECAUSE it's a bulky item(also because it's fairly rare). Making it normal sized makes it way more convenient for anyone to lug one around, where as at the moment you have to get an ion rifle and a BoH to make it work, which has it's own drawbacks due to teleport fuckery.

**Comment by GunHog** [comment]
PR updated in compliance with Paprika's new flashlight system.

**Comment by Miauw62** [comment]
i'd like to note that EMP weapons also completely wreck silicons, like, seriously. it's effectively a hard counter.

**Comment by GunHog** [comment]
I think requiring at least three departments to work together to acquire this upgrade is sufficient. Supply must succeed in acquiring the minerals, someone must be manning Science and handling research, and Security/Captain must provide firing pins or the weapon must be given to them. Antagonists can bypass Command/Sec by stealing pins or emagging a test range pin, but that is a different situation.

I feel it is still in line to have Science (Borg console, flashes, upgrades for Sec and others) and Engineering (nigh-invulnerable to the AI itself) as the primary nightmares for Silicons. Sec is super effective against all other antags, and if they want to truly be *anti-everything*, they should have to swallow their pride of being completely self-sufficient and actually work with other departments for a change.

File: code/modules/projectiles/guns/energy/special.dm [hunk 1]
```diff
@@ -132,7 +142,7 @@
 	if(!suppressed)
 		playsound(src.loc, 'sound/weapons/kenetic_reload.ogg', 60, 1)
 	else
-		user << "<span class='warning'>You silently charge [src].<span>"
+		playsound(user, 'sound/weapons/kenetic_reload.ogg', 60, 1)
```
Comment (phil235) [inline code/modules/projectiles/guns/energy/special.dm]: What's the point of the if(!suppressed) if both cases do the same thing?

Comment (GunHog) [inline code/modules/projectiles/guns/energy/special.dm]: How in the? This does not belong here. Clearly I fudged the merge.

**Pull request merged**